UPGRADE_GUIDE.md

```diff
@@ -2,9 +2,150 @@
+**Dependencies**
+
+All dependencies had major and minor version bumps. All previous security vulnerabilities have been remedied.
+
+## 6.0 Create and Update Functions
+
+Likelihood of Impact: High
+
+Previously, one could create a local object and then call `.save()` on it which would either create that object or update it via the API. This is no longer possible.
+
+Instead of creating or updating local objects via a save function, there are new `create()` and `update()` functions. You will call these functions against a "service" like you can with `retrieve` or `all`.
+
+```javascript
+// Old approach (save called against a local object)
+const api = new Easypost(process.env.EASYPOST_API_KEY);
+const parcel = new api.Parcel({
+  length: 20.2,
+  width: 10.9,
+  height: 5,
+  weight: 65.9,
+});
+parcel.save().then(console.log);
+
+// New approach (save called against the Parcel service of a client object)
+const client = new Easypost(process.env.EASYPOST_API_KEY);
+const parcel = await client.Parcel.create({
+  length: 20.2,
+  width: 10.9,
+  height: 5,
+  weight: 65.9,
+});
+console.log(parcel);
+```
+
+## 6.0 Removal of Instance Methods
+
+All instance methods have been removed. This includes things like `.save()`, `.delete()`, `.buy()`, etc. You now call these methods against a service and pass in the ID of the object in question. This change was made to further support thread-safety efforts with differing API keys along with ensuring that data flows properly through the associated service and not through local objects.
+
+The `.lowestRate()` method of a Shipment, Order, or Pickup is an error and is still valid.
+
+```javascript
+// Old approach (buy function called on a local shipment object)
+const api = new Easypost(process.env.EASYPOST_API_KEY);
+api.Shipment.retrieve('shp_123...').then((shipment) => {
+  shipment.buy(shipment.lowestRate()).then(console.log);
+});
+
+// New approach (buy function called against the Shipment service, of a client object, with the shipment ID passed in)
+const client = new Easypost(process.env.EASYPOST_API_KEY);
+const shipment = await client.Shipment.retrieve('shp_123...');
+const boughtShipment = client.Shipment.buy(shipment.id, shipment.lowestRate());
+console.log(boughtShipment);
+```
+
+All functions attached to a service are now async and must be awaited.
+
+Services now implement all their own functions instead of inheriting from the BaseService meaning the library's namespace will no longer be cluttered with various `notImplemented` functions
+
+## 6.0 Overhauled Error Handling
+
+Introduced ~2 dozen new error types that extend from either `ApiError` or `EasyPostError`.
+
+New ApiError (extends EasyPostError):
+
+- `errors/api/api_error`
+- `errors/api/encoding_error`
+- `errors/api/external_api_error`
+- `errors/api/forbidden_error`
+- `errors/api/gateway_timeout_error`
+- `errors/api/http_error`
+- `errors/api/internal_server_error`
+- `errors/api/invalid_request_error`
+- `errors/api/json_error`
```

Comment:
Is JSON error an API error or a local error? Writing the docstring now, and it's not used anywhere. In C#, this error is for when de/serialization fails locally, but not sure if it's the same (or even needed) in Node.

Reply:
Probably not in Node, I should have excluded this for Node

Reply:
I'll delete it in my docstrings PR

Reply:
Ok, so wanna stamp this one? 😉